scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+    # `**kwargs`, the parameter dictionary passed from the public method via
+    # the "dispatch" method.
+    # See the note corresponding with the "Distribution Parameters" for more
+    # information.
+
+    def sample(self, shape=(), *, method=None, rng=None, qmc_engine=None):
+        """Random or quasi-Monte Carlo sample from the distribution
+
+        Parameters
+        ----------
+        shape : tuple of ints, default: ()
+            The shape of the sample to draw. If the parameters of the distribution
+            underlying the random variable are arrays of shape ``param_shape``,
+            the output array will be of shape ``shape + param_shape``.
+        method : {None, 'formula', 'inverse_transform'}
+            The strategy used to produce the sample. By default (``None``),
+            the infrastructure chooses between the following options.
+
+            - ``'formula'``: an implementation specific to the distribution
+            - ``'inverse_transform'``: generate a uniformly distributed sample and
+                                       return the inverse CDF at these arguments.
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a `NotImplementedError``
+            will be raised.
+        rng : `numpy.random.Generator`, optional
+            The pseudorandom number generator instance with which to generate
+            the sample. If `None` (default), a new generator is instantiated
+            with fresh, unpredictable entropy from the operating system.
+        qmc_engine : `scipy.stats.qmc.QMCEngine` subclass, optional
+            A QMC engine class with which to generate a quasi-Monte Carlo sample.
+            An instance of the `qmc_engine` class will be created and provided
+            with `rng` (e.g. for use with shuffling). Typically, the use of
+            `qmc_engine` with ``method='formula'`` will be incompatible.
+
+        Notes
+        -----
+        The values of a quasi-Monte Carlo sequence are not statistically
+        independent; the sequence is designed to have low-discrepancy.
+        The output of `sample` is always formed with this low-discrepancy
+        sequence aligned along axis ``0``. Separate slices along axis ``0``
+        (e.g. separate columns of a 2D output) are separate low-discrepancy
+        sequences; the low-discrepancy properties hold *only* along axis
+        ``0``, not along other axes. By default, many `QMCEngine` classes
+        use a pseudorandom number generator (provided by `rng`, in this case)
+        to *scramble* these separate sequences so they are statistically
+        independent. Consequently, the following simple rule holds for
+        quasi-Monte Carlo samples drawn using `QMCEngine` classes that
+        employ scrambling: each slice along axis ``0`` of the result is a
+        statistically independent low-discrepancy sequence.
```

Comment:
This sounds misleading to me. If you create a QMC sequence in n-dimensions, then the low discrepancy properties hold not only on single axis/dim, but along any axis/dim combinations. When I read the above, I have the impression that you are saying that you use 1D QMC samples and concatenate to have multiple dims. But then in the code you instantiate with multiple dimensions and not just 1. So I am not sure what exactly you want to say.

In the end, my recommendation would be to simply link to the QMC doc. IIRC, this is what we did for Christoph's samplers.

Reply:
> This sounds misleading to me.

I think this is what I intended but the implementation does not give us that. 

If we use a multi-dimensional QMCEngine to produce the samples, I think the discrepancy along a given axis will tend to be higher than if we use multiple samples from a 1-D QMC Engine, right?

```python3
import numpy as np
from scipy.stats.qmc import Halton, discrepancy
qrng = Halton(10)
x = qrng.random(10)
print([discrepancy(xi[:, np.newaxis]) for xi in x])
print([discrepancy(xi[:, np.newaxis]) for xi in x.T])
# [0.009951136994696075, 0.008858673211191936, 0.01476610903829223, 0.023091188123706363, 0.05905334311169752, 0.040271487656198035, 0.06212596060712716, 0.0336833682094575, 0.0020552732026202225, 0.017731702680111594]
# [0.0019172673788583339, 0.0026253028968135084, 0.001149728179166143, 0.002827787052197772, 0.0016142233504383707, 0.007647829657824401, 0.025791264610144138, 0.0029659741646828053, 0.020724357197512644, 0.0034924829972200477]
```
VS
```python3
qrng = Halton(1)
print([discrepancy(qrng.random(10)) for i in range(10)])
# [0.0017679568590851513, 0.002598484920673716, 0.0058226494587596545, 0.00419583951305702, 0.0017254405955107188, 0.0033288028418654303, 0.002814797339178998, 0.0014657120402539192, 0.001919138697870748, 0.003263486913431146]
# all less than 0.01
```

Since this is for a *univariate* infrastructure, I think we'd want to keep the discrepancy low along a single axis, right? If the user wants low-discrepancy in a high-dimensional space, they would need a multivariate distribution.

Reply:
Ok I see what you mean now. Yes a multi-dim `QMCEngine` will have a lower discrepancy on each of its individual dimension vs using a 1D engine.

I was confused because the text seemed to say that a sample in n-D would have a higher discrepancy than using n-1D samples concatenated together. But this is not what we do as we just want to sample n times a 1D sample. I think just saying that would be clearer.